frontstage/views/surveys.py

```diff
@@ -1,37 +1,164 @@
 import json
 import logging
+from os import getenv
 
 from flask import Blueprint, redirect, render_template, request, url_for
 from structlog import wrap_logger
 
 from frontstage import app
 from frontstage.common.api_call import api_call
 from frontstage.common.authorisation import jwt_authorization
+from frontstage.common.cryptographer import Cryptographer
 from frontstage.exceptions.exceptions import ApiError
-
+from frontstage.models import EnrolmentCodeForm
 
 logger = wrap_logger(logging.getLogger(__name__))
 surveys_bp = Blueprint('surveys_bp', __name__,
                        static_folder='static', template_folder='templates/surveys')
+cryptographer = Cryptographer()
 
 
 @surveys_bp.route('/', methods=['GET'])
 @jwt_authorization(request)
 def logged_in(session):
     party_id = session.get('party_id')
     surveys_list = get_surveys_list(party_id, 'todo')
-    return render_template('surveys/surveys-todo.html', _theme='default', surveys_list=surveys_list)
+    template_data = {"survey": {"justAdded": "false"}}
+
+    sorted_surveys_list = sorted(surveys_list, key=lambda k: k['collection_exercise']['scheduledReturnDateTime'],
+                                 reverse=True)
+
+    # Checks if new survey is added and renders "survey added" notification
+    if request.args.get('new_survey') and ['NEW_SURVEY_NOTIF_HIGHLIGHTING'] == 1:
+        logger.info('New survey added highlighted')
+        template_data = {"survey": {"justAdded": "true"}}
+        return render_template('surveys/surveys-todo.html', _theme='default',
+                               data=template_data, sorted_surveys_list=sorted_surveys_list,
+                               just_added_case_id=request.args.get('case_id'))
+
+    return render_template('surveys/surveys-todo.html', data=template_data,
+                           _theme='default', sorted_surveys_list=sorted_surveys_list)
 
 
 @surveys_bp.route('/history', methods=['GET'])
 @jwt_authorization(request)
 def surveys_history(session):
     party_id = session['party_id']
     surveys_list = get_surveys_list(party_id, 'history')
-    return render_template('surveys/surveys-history.html',  _theme='default',
+    template_data = {"survey": {"justAdded": "false"}}
```

Comment:
If we do as Tej recommended above and have `survey_added` be a boolean we could leave this unchanged